[devops] Publish mlaunch as .nupkg during main build

premun opened the pull request:
Packages the mlaunch that was built during the pipeline previously as a .nupkg and publishes it into the [dotnet-eng feed](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=dotnet-eng):
- Version contains the maccore commit SHA
- If this SHA is already published in the feed, packaging and publishing is skipped

Target package: https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet-eng&package=Microsoft.DotNet.Mlaunch&protocolType=NuGet (will show up after main build completes)

https://github.com/dotnet/core-eng/issues/13150

vs-mobiletools-engineering-service2 commented:
### :fire: Tests failed catastrophically on Build (no summary found). :fire:

Result file $(TEST_SUMMARY_PATH) not found. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4847447&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/sim/;/tests/vsdrops_index.html)
# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11851/4847447/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11851/4847447/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11851/4847447/apicomparison/api-diff.html)


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4847447&view=ms.vss-test-web.test-result-details) on Agent 
Merge 6832026e7a8abadc0592441430880854f0f03a82 into 3d9c37778f0ea81fdb02ce16ed0ecefb9cc13152

rolfbjarne commented on tools/devops/automation/scripts/bash/build-nugets.sh:
```diff
@@ -18,3 +23,42 @@ DOTNET_PKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_PKG
 cp -c "$DOTNET_PKG_DIR"/*.pkg ../package/
 cp -c "$DOTNET_PKG_DIR"/*.msi ../package/
+
+# Down from here we publish mlaunch as .nupkg
```
I think this logic could go into its own shell script (say `build-mlaunch-nuget.sh`)

rolfbjarne commented on tools/devops/automation/scripts/bash/build-nugets.sh:
```diff
@@ -18,3 +23,42 @@ DOTNET_PKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_PKG
+MONOTOUCH_PREFIX=$(make -C tools/devops print-abspath-variable VARIABLE=MONOTOUCH_PREFIX | grep "^MONOTOUCH_PREFIX=" | sed -e 's/^MONOTOUCH_PREFIX=//')
+
+# Copy mlaunch to staging area
+cp -r "$MACCORE_TOP/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/bin/Debug/mlaunch.app" "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
+cp "$IOS_DESTDIR$MONOTOUCH_PREFIX/bin/mlaunch" "$MLAUNCH_WORK_DIR/mlaunch/bin/"
```
Cloning is faster and uses less disk space:

```suggestion
cp -c -r "$MACCORE_TOP/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/bin/Debug/mlaunch.app" "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
cp -c "$IOS_DESTDIR$MONOTOUCH_PREFIX/bin/mlaunch" "$MLAUNCH_WORK_DIR/mlaunch/bin/"
```

rolfbjarne commented on tools/devops/automation/scripts/bash/build-nugets.sh:
```diff
@@ -18,3 +23,42 @@ DOTNET_PKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_PKG
+cp "$IOS_DESTDIR$MONOTOUCH_PREFIX/bin/mlaunch" "$MLAUNCH_WORK_DIR/mlaunch/bin/"
+
+# Add the .csproj we will use to create the .nupkg
+cp "$XAM_TOP/tools/mlaunch/Microsoft.DotNet.Mlaunch.csproj" "$MLAUNCH_WORK_DIR"
```
```suggestion
cp -c "$XAM_TOP/tools/mlaunch/Microsoft.DotNet.Mlaunch.csproj" "$MLAUNCH_WORK_DIR"
```

rolfbjarne commented on tools/devops/automation/scripts/bash/build-nugets.sh:
```diff
@@ -18,3 +23,42 @@ DOTNET_PKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_PKG
+cp "$XAM_TOP/tools/mlaunch/Microsoft.DotNet.Mlaunch.csproj" "$MLAUNCH_WORK_DIR"
+
+# We need to override global.json to use .NET 6.0
+cp "$XAM_TOP/global6.json" "$MLAUNCH_WORK_DIR/global.json"
```
```suggestion
cp -c "$XAM_TOP/global6.json" "$MLAUNCH_WORK_DIR/global.json"
```

rolfbjarne commented on tools/devops/automation/scripts/bash/build-nugets.sh:
```diff
@@ -18,3 +23,42 @@ DOTNET_PKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_PKG
+# We store mlaunch NuGet in [build work root]/mlaunch
+cd "$XAM_TOP"
+mkdir ../mlaunch
+cp "$MLAUNCH_WORK_DIR"/*.nupkg ../mlaunch/
```
```suggestion
cp -c "$MLAUNCH_WORK_DIR"/*.nupkg ../mlaunch/
```

rolfbjarne commented on tools/mlaunch/Microsoft.DotNet.Mlaunch.csproj:
```diff
@@ -0,0 +1,17 @@
+		<TargetFramework>net6.0</TargetFramework>
+		<!-- Suppresses the warnings about the package not having assemblies in lib/*/.dll.-->
+		<NoPackageAnalysis>true</NoPackageAnalysis>
+		<VersionPrefix>1.0.0</VersionPrefix>
```
I think we should come up with a better version than 1.0.0.

We could use the existing mlaunch version:

```shell
$ /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mlaunch --version 
Xamarin.Launcher 14.21.0.330 (xamarin/main: cddbd1915d)
```

or we could come up with something else (Xcode version? Something completely different that's calculated in maccore and doesn't take xamarin-macios into account? Go with "0.0.1" and decide something later?)

@spouliot / @dalexsoto what do you think? I lean towards the mlaunch version (although the downside here is that there can be multiple mlaunch versions for the same maccore hash).

spouliot replied:
IMO it would be useful to use the Xcode version for which `mlaunch` was built (and known to work [1]) for. We should also print that version (at least for the nuget version, maybe not the one we ship bundled in the legacy SDK)

[1] and _should_ work on earlier Xcode too...

Using legacy version is confusing as 14.21 means almost nothing, except iOS 14.x -> Xcode 12.x
That already requires mapping (-2) but only works for the major version.

Using 1.0.0 would require a new totally mapping, ending up in more confusion.

spouliot replied:
I still see 1.0.0 here ?

premun replied:
@spouliot there's a default that needs to be in (for example if you build it locally). We override that in the CI:
```bash
"$DOTNET6" pack --version-suffix "$MACCORE_HASH" /p:VersionPrefix=$XCODE_VERSION
```

vs-mobiletools-engineering-service2 commented:
### :white_check_mark: [PR Build] Tests passed on Build. :white_check_mark:

Tests passed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4847447&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/7fe1ad5981a02b8150d03e684d04fe49)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/7d8aea244150f7a8eb018850239885e4)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/ef713ad1c4c0d8bd22492f49f9ce3c9f)
* macCatiOS  [gist](https://gist.github.com/c1eaa1ea62c9f8adb84aab9e8320b72f)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/02b8661a2c3c58907a8ddb2496e828a1)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/b4bb2edc0023479fb5119303a204b621)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210608.8/4847447/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11851/4847447/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11851/4847447/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11851/4847447/apicomparison/api-diff.html)


# :tada: All 81 tests passed :tada:



[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4847447&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1104.BigSur'
Merge 6832026e7a8abadc0592441430880854f0f03a82 into 3d9c37778f0ea81fdb02ce16ed0ecefb9cc13152

rolfbjarne commented on tools/devops/automation/scripts/bash/build-mlaunch-nuget.sh:
```diff
@@ -0,0 +1,56 @@
+# We need to override global.json to use .NET 6.0
+cp -c "$XAM_TOP/global6.json" "$MLAUNCH_WORK_DIR/global.json"
+
+version=$("$MLAUNCH_WORK_DIR/mlaunch/bin/mlaunch" --version | cut -d " " -f 2)
```
Let's do what Sebastien said, and use the Xcode version (the XCODE_VERSION variable in Make.config).

premun replied:
Done!

premun commented:
I have one more change I want to add - I want to add some NuGet hooks so that I will be able to specify something like this:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <MlaunchDestinationDir>$(IntermediateOutputPath)\mlaunch</MlaunchDestinationDir>
  </PropertyGroup>
  
  <ItemGroup>
    <PackageReference Include="Microsoft.DotNet.Mlaunch" Version="12.5-cdbefac" />
  </ItemGroup>

</Project>
```

And the NuGet will copy the `mlaunch` folder there before Build.

@rolfbjarne should I add this as part of this PR or separate?

rolfbjarne commented:
@premun that can be a separate PR once this is in

premun commented:
The current package name is `Microsoft.DotNet.Mlaunch.12.5.0-cddbd1915d.nupkg`

If the build goes green, we can merge?

vs-mobiletools-engineering-service2 commented:
### :white_check_mark: [PR Build] Tests passed on Build. :white_check_mark:

Tests passed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4852787&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/c1f4e590e68152b682bac73b34bc3c0c)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/2b91e5e699e13aa76aca462e7af94c0c)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/a9f0dfea60a108c9889c9cdef8e2e2dc)
* macCatiOS  [gist](https://gist.github.com/180ddd694690b2c68a853a64600b48e0)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/229ff9125b841999b334b1da1ea9bcd8)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/4fe13d8196763f71f7db6a83fe250953)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.27/4852787/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11851/4852787/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11851/4852787/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11851/4852787/apicomparison/api-diff.html)


# :tada: All 81 tests passed :tada:



[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4852787&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1104.BigSur'
Merge ab13424e10640d06be4f2933668b6522549bddcd into 7916e74b89c8cf4fb30a184697e37e989cd6d669

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4854321&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/a40a40b811548419c775432a521373d4)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/6541957684e49bb5faa5c3b068e5ec74)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/5b5e7bd1e66e0eea58b854c00778ee4f)
* macCatiOS  [gist](https://gist.github.com/7f22bb81f89a48c5f991e5fb812d515a)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/7f2d5926544d532ac51e17572541a3cc)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/684a59b9fa35100ed88c2696f927fa88)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210609.41/4854321/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11851/4854321/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11851/4854321/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11851/4854321/apicomparison/api-diff.html)


# Test results

<details>
<summary>1 tests failed, 80 tests passed.</summary>

## Failed tests

 * monotouch-test/iOS Unified 64-bits - simulator/Debug: Failed
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4854321&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1104.BigSur'
Merge 7bc8ec106e606f692f434d605d1f7ff5780ac846 into 233497d79189c84e0fc38a25952d2a6d3f476078

spouliot commented on tools/devops/automation/scripts/bash/build-mlaunch-nuget.sh:
```diff
@@ -0,0 +1,55 @@
+
+# We have to build from within the dir to respect the global.json
+cd "$MLAUNCH_WORK_DIR"
+"$DOTNET6" pack --version-suffix "$MACCORE_HASH" /p:VersionPrefix=$XCODE_VERSION
```
Sadly we might need more than one mlaunch release for each version of Xcode, bug happens :(

> The current package name is Microsoft.DotNet.Mlaunch.12.5.0-cddbd1915d.nupkg

If you do an update (for Xcode 12.5.0) then only the digest (cddbd1915d) will change. 
It will still be a unique name but no one will be able to figure out which is the newest version.

Something more (like the commit distance or however net6 compute `21276.13`) is needed to distinguish the published packages.

premun replied:
I searched Arcade and we use this formula in .NET 6:
```
(yy * 1000 + mm * 50 + dd).(BuildRevision)
```

So today's number is:
```
21 * 1000 + 06 * 50 + 10 = 21310.x
```

source: https://github.com/dotnet/arcade/blob/e81d8c9bdc1de22623657afb23fab210dd04ca82/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets#L57

So I can do `12.5.0-21310.1-cddbd1915d`?
Or we rather want `21310.1-12.5.0-cddbd1915d`?

premun replied:
Or I have seen in Xamarin previously `+cddbd1915d` or `+sha_cddbd1915d` I think

rolfbjarne replied:
I think something like this could work:

> 12.5.0-21310.1.cddbd1915d

premun replied:
So, we now have `Microsoft.DotNet.Mlaunch.12.5.0-21310.9.cddbd1915d.nupkg`

Built here:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4857230&view=logs&j=600d49c6-e96b-5e8b-2c3a-9f351e2e5848&t=c9117ceb-ee46-564f-0d31-c57e59243b74&l=181

premun commented:
@rolfbjarne I added the feature with the extraction because I wasn't able to merge, had the changes in my branch and wouldn't be able to use the mlaunch without it anyway. It's not a lot of code and I added a README explaining the usage to the NuGet:
https://devblogs.microsoft.com/nuget/add-a-readme-to-your-nuget-package/

Hope it's fine

vs-mobiletools-engineering-service2 commented:
### :white_check_mark: [PR Build] Tests passed on Build. :white_check_mark:

Tests passed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4857230&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/285cc4db874103ba6204f4255ab9d6bb)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/42454662f3e9a75d103e40c00cd1904a)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/92d89430b65c3666c44daf510a8e0636)
* macCatiOS  [gist](https://gist.github.com/c0fcd59889e2cf10af8ddb5719b9d5b3)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/18aadd39678fba1f47d286c088724206)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/c006b90867c1df61630c7ff4db2a1865)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210610.9/4857230/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11851/4857230/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11851/4857230/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11851/4857230/apicomparison/api-diff.html)


# :tada: All 81 tests passed :tada:



[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4857230&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1094.BigSur'
Merge 230161e63b259d1a810ece11472affefc647c398 into 8ca5d5ea5c062093220cf289f1992edbcb45dcec
